src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_InvalidStatement.vb

```diff
@@ -0,0 +1,292 @@
+ISwitchStatement (1 cases) (OperationKind.SwitchStatement, IsInvalid)
+  Switch expression: IInvocationExpression (virtual Function System.Object.ToString() As System.String) (OperationKind.InvocationExpression, Type: System.String)
+      Instance Receiver: ILocalReferenceExpression: x (OperationKind.LocalReferenceExpression, Type: Program)
+  ISwitchCase (1 case clauses, 1 statements) (OperationKind.SwitchCase, IsInvalid)
+    Case clauses: ISingleValueCaseClause (Equality operator kind: BinaryOperationKind.Invalid) (CaseKind.SingleValue) (OperationKind.SingleValueCaseClause, IsInvalid)
+    Body: IBlockStatement (1 statements) (OperationKind.BlockStatement)
+        IBranchStatement (BranchKind.Break, Label: exit) (OperationKind.BranchStatement)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of SelectBlockSyntax)(source, expectedOperationTree)
+        End Sub
+
+        <Fact, WorkItem(17607, "https://github.com/dotnet/roslyn/issues/17607")>
+        Public Sub InvalidIfStatement()
+            Dim source = <![CDATA[
+Class Program
+    Private Shared Sub Main(args As String())
+        Dim x = New Program()
+        If x = Nothing Then'BIND:"If x = Nothing Then"
+        End If
+    End Sub
+End Class
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IIfStatement (OperationKind.IfStatement, IsInvalid)
+  Condition: IConversionExpression (ConversionKind.Basic, Implicit) (OperationKind.ConversionExpression, Type: System.Boolean, IsInvalid)
+      IBinaryOperatorExpression (BinaryOperationKind.Invalid) (OperationKind.BinaryOperatorExpression, Type: ?, IsInvalid)
+        Left: ILocalReferenceExpression: x (OperationKind.LocalReferenceExpression, Type: Program)
+        Right: IConversionExpression (ConversionKind.Basic, Implicit) (OperationKind.ConversionExpression, Type: Program, Constant: null)
+            ILiteralExpression (OperationKind.LiteralExpression, Type: null, Constant: null)
+  IBlockStatement (0 statements) (OperationKind.BlockStatement)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of MultiLineIfBlockSyntax)(source, expectedOperationTree)
+        End Sub
+
+        <Fact, WorkItem(17607, "https://github.com/dotnet/roslyn/issues/17607")>
+        Public Sub InvalidIfElseIfStatement()
+            Dim source = <![CDATA[
+Class Program
+    Private Shared Sub Main(args As String())
+        Dim x = New Program()
+        If Then'BIND:"If Then"
+        ElseIf x Then
+            x
+        Else
+    End Sub
+End Class
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IIfStatement (OperationKind.IfStatement, IsInvalid)
+  Condition: IConversionExpression (ConversionKind.Basic, Implicit) (OperationKind.ConversionExpression, Type: System.Boolean, IsInvalid)
+      IInvalidExpression (OperationKind.InvalidExpression, Type: ?, IsInvalid)
+  IBlockStatement (0 statements) (OperationKind.BlockStatement)
+  IIfStatement (OperationKind.IfStatement, IsInvalid)
```

Comment:
I will enhance the dumper to make the associated syntax and child properties explicit.